pisa/stages/osc/prob3.py

```diff
@@ -360,12 +393,39 @@ def compute_function(self):
                 self.params.eps_mutau_phase.value.m_as('rad'))
             )
             self.nsi_params.eps_tautau = self.params.eps_tautau.value.m_as('dimensionless')
-        
         if self.neutrino_decay:
             self.decay_params.decay_alpha3 = self.params.decay_alpha3.value.m_as('eV**2')
             
         if self.lri_type is not None:
             self.lri_params.v_lri = self.params.v_lri.value.m_as('eV')
+        if self.tomography_type == "mass_of_earth":
+                self.tomography_params.density_scale = self.params.density_scale.value.m_as('dimensionless')
+                self.layers = Layers(earth_model, detector_depth, prop_height,scaling_array=self.tomography_params.density_scale)
```

Comment:
I think it would be better to give the `Layers` class something like a scale function that you can call here to not initialize the class every time the stage is called. That would simplify the entire code (not only this part) and should be faster.